mesonbuild/modules/qt.py

```diff
@@ -456,6 +456,7 @@ def _compile_moc_impl(self, state: 'ModuleState', kwargs: 'MocCompilerKwArgs') -
         compile_args: T.List[str] = []
         for dep in kwargs['dependencies']:
             compile_args.extend([a for a in dep.get_all_compile_args() if a.startswith(('-I', '-D'))])
+            compile_args.extend([a.replace('-isystem', '-I', 1) for a in dep.get_all_compile_args() if a.startswith(('-isystem'))])
```

Comment:
Do we need to reformat the args here? That is, does moc understand `-isystem` flags?

Reply:
1. Actually my code is bad, according to clang documentation after '-isystem' can be a space symbol. So it have to be much more complicated, and can take two positions in `compile_args` (space between `-isystem` and actual path accepted by both clang and gcc, don't test on others compilers).
2. Also according clang [documentation](https://clang.llvm.org/docs/ClangCommandLineReference.html#include-path-management) there can be few more include schemes. Btw, in gcc the flags are different, and I can't find good documentation on it (i mean with some kind of bnf).
3. `moc` does not understand `-isystem` flag. And even worse, when compiling under windows you have to add `-D_WIN32` (and `-DWIN32` if you need it).

So... May be a better solution would be to say `Don't do that, provide only '-I' includes to moc`.
I don't know which decision is better, what have we do.
Possible way is to change documentation on `qt.preprocess`, and state all problems with it.